### PR TITLE
render the workload and trait name differently

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
Special treatment for names in appConfig generated by an Application
1. in-place upgradable workload name will be component 
2. otherwise, it will be componentRevision Name
3. We ban name in the metadata field

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go [hunk 1]
```diff
@@ -342,6 +322,30 @@ func isRevisionEnabled(traitDefs []v1alpha2.TraitDefinition) bool {
...
+// setAppWorkloadInstanceName sets the name of the workload instance depends on the component revision
+// and the workload kind
+func setAppWorkloadInstanceName(componentName string, w *unstructured.Unstructured, revision int) {
+	// TODO: we can get the workloadDefinition name from w.GetLabels()["oam.WorkloadTypeLabel"]
+	// and use a special field like "use-inplace-upgrade" in the definition to allow configurable behavior
+
+	// we hard code the behavior depends on the workload group/kind for now. The only in-place upgradable resources
+	// we support is cloneset for now. We can easily add more later.
+	if w.GroupVersionKind().Group == kruise.GroupVersion.Group {
+		if w.GetKind() == reflect.TypeOf(kruise.CloneSet{}).Name() {
```
Comment (zzxwill) [inline pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go:334]: Cool.

Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go:334]: also consider statefulset

**Review by wonderflow — APPROVED** [review]
good job

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go (hunk 1 above)
Comment (ryanzhang-oss) [reply pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go:334]: I was going to but not 100% sure if statefulset should be upgraded in-place

**Pull request merged**